spyder/widgets/mixins.py

```diff
@@ -68,9 +68,12 @@ def calculate_real_position(self, point):
     #------Calltips
     def _format_signature(self, text):
         formatted_lines = []
-        name = text.split('(')[0]
-        rows = textwrap.wrap(text, width=50,
-                             subsequent_indent=' '*(len(name)+1))
+        # This code was assuming the text included the signature, which is
+        # no longer the case
+        # name = text.split('(')[0]
+        # rows = textwrap.wrap(text, width=50,
+        #                      subsequent_indent=' '*(len(name)+1))
+        rows = text.split('\n')
         for r in rows:

```

Comment:
This was assuming the signature was included in the text which is no longer the case. But probably all the styling that is being made on the code editor, should be made here.

Thougths?

Reply:
I think this part could be important to maintain it

> rows = textwrap.wrap(text, width=50, subsequent_indent=' '*(len(name)+1))

It wraps the signature to make it 50 chars long, which avoids displaying a huge calltip for objects with a lot of arguments. You can try with `subprocess.Popen` in Python 3 to see what I mean.